python/tests/catalog/test_rest.py

```diff
@@ -0,0 +1,670 @@
+from pyiceberg.table.sorting import SortField, SortOrder
+from pyiceberg.transforms import IdentityTransform, TruncateTransform
+from pyiceberg.types import (
+    BooleanType,
+    IntegerType,
+    NestedField,
+    StringType,
+)
+
+TEST_HOST = "https://iceberg-test-catalog/"
+TEST_TOKEN = "client:secret"
+
+
+def test_token_200():
+    with requests_mock.Mocker() as m:
+        m.post(
+            f"{TEST_HOST}oauth/tokens",
+            json={
+                "access_token": TEST_TOKEN,
+                "token_type": "Bearer",
+                "expires_in": 86400,
+                "warehouse_id": "8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+                "region": "us-west-2",
+                "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+            },
+            status_code=200,
+        )
+        m.get(
+            f"{TEST_HOST}config",
+            json={"defaults": {}, "overrides": {"prefix": "oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e"}},
+            status_code=200,
+        )
+
+        assert RestCatalog("rest", {}, TEST_HOST, TEST_TOKEN).token == TokenResponse(
+            access_token=TEST_TOKEN,
+            token_type="Bearer",
+            expires_in=86400,
+            warehouse_id="8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+            region="us-west-2",
+            issued_token_type="urn:ietf:params:oauth:token-type:access_token",
+        )
+
+
+def test_token_401():
+    with requests_mock.Mocker() as m:
+        message = "Invalid client ID: abc"
+        m.post(
+            f"{TEST_HOST}oauth/tokens",
+            json={
+                "error": {
+                    "message": message,
+                    "type": "BadCredentialsException",
+                    "code": 401,
+                }
+            },
+            status_code=401,
+        )
+
+        with pytest.raises(BadCredentialsError) as e:
+            RestCatalog("rest", {}, TEST_HOST, TEST_TOKEN)
+        assert message in str(e.value)
+
+
+def _configure_token_and_config_endpoint(m: requests_mock.Mocker):
+    m.post(
+        f"{TEST_HOST}oauth/tokens",
+        json={
+            "access_token": "eyJ0eXAiOiJK",
+            "token_type": "Bearer",
+            "expires_in": 86400,
+            "warehouse_id": "8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+            "region": "us-west-2",
+            "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+        },
+        status_code=200,
+    )
+    m.get(
+        f"{TEST_HOST}config",
+        json={"defaults": {}, "overrides": {"prefix": "oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e"}},
+        status_code=200,
+    )
+
+
+def test_list_tables_200():
+    with requests_mock.Mocker() as m:
+        _configure_token_and_config_endpoint(m)
+        namespace = "examples"
+        m.get(
+            f"{TEST_HOST}oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e/namespaces/{namespace}/tables",
```

Comment:
I think the tests should primarily use an empty prefix. There's no need for an arbitrary prefix in these cases.

This is one reason why the Java side uses a class to build URLs. We can test the prefix behavior of that class without testing whether every catalog method supports `prefix` correctly.

Reply:
I've removed the prefix.